src/builder/commands/composer.rs

```diff
@@ -0,0 +1,269 @@
+
+    ctx.environ.insert("COMPOSER_HOME".to_owned(),
+                       COMPOSER_HOME.to_owned());
+    ctx.environ.insert("COMPOSER_VENDOR_DIR".to_owned(),
+                       COMPOSER_VENDOR_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_BIN_DIR".to_owned(),
+                       COMPOSER_BIN_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_CACHE_DIR".to_owned(),
+                       COMPOSER_CACHE.to_owned());
+
+    Ok(())
+}
+
+pub fn bootstrap(ctx: &mut Context) -> Result<(), String> {
+    try_msg!(file_util::create_dir(COMPOSER_HOME, true),
+        "Error creating composer home dir {d:?}: {err}", d=COMPOSER_HOME);
+
+    let composer_inst = try!(download::download_file(ctx, COMPOSER_BOOTSTRAP));
+    try!(file_util::copy(&composer_inst, &Path::new("/vagga/root/tmp/composer-setup.php"))
+        .map_err(|e| format!("Error copying composer installer: {}", e)));
+
+    let runtime_exe = ctx.composer_settings
+        .runtime_exe
+        .clone()
+        .unwrap_or(DEFAULT_RUNTIME.to_owned());
+
+    let args = [
+        runtime_exe,
+        "/tmp/composer-setup.php".to_owned(),
+        "--install-dir=/tmp/".to_owned(),
+    ];
+    try!(run_command(ctx, &args));
+
+    try_msg!(unix_fs::symlink("/vagga/root/tmp/composer.phar", "/vagga/root/usr/local/bin/composer"),
+        "Error creating symlink '/usr/local/bin/composer -> /tmp/composer.phar': {err}");
+
+    if ctx.composer_settings.install_runtime {
+        try!(setup_include_path(ctx));
+    }
+
+    Ok(())
+}
+
+fn setup_include_path(ctx: &mut Context) -> Result<(), String> {
+    let vagga_ini_content = {
+        let include_path = ctx.composer_settings
+            .include_path.clone()
+            .unwrap_or(DEFAULT_INCLUDE_PATH.to_owned());
+        format!("include_path={}", include_path)
+    };
+
+    // find conf.d directories (various can exist on Ubuntu)
+    let conf_dirs = try!(find_dirs(ctx, "/etc/php**/conf.d"));
+
+    for conf_d in conf_dirs.iter() {
+        // create vagga.ini file
+        try!(create_vagga_ini(&conf_d.join("vagga.ini"), &vagga_ini_content));
+    }
+
+    if !conf_dirs.is_empty() {
+        // vagga.ini file(s) created, we're done here
+        return Ok(())
+    }
+
+    // If we didn't find any conf.d, ask 'php --ini'
+    let conf_d = try!(ask_php_for_conf_d(ctx));
+
+    // create conf.d
+    if !conf_d.exists() {
+        try!(file_util::create_dir(&conf_d, true)
+        .map_err(|e| format!("Error creating directory {:?}: {}", conf_d, e)));
+    }
+    // create vagga.ini file
+    try!(create_vagga_ini(&conf_d, &vagga_ini_content));
+
+    Ok(())
+}
+
+fn create_vagga_ini(location: &Path, content: &str) -> Result<(), String> {
+    File::create(location)
+        .and_then(|mut f| f.write_all(content.as_bytes()))
+        .map_err(|e| format!("Error creating file {:?}: {}", location, e))
+}
+
+fn find_dirs(ctx: &mut Context, search: &str) -> Result<Vec<PathBuf>, String> {
```

Comment:
Okay. Why recursive? Just walk through `/etc/php/*` and look for `conf.d` in each of them.